app/api/auth/oauth.py

```diff
@@ -0,0 +1,66 @@
+# app/auth/oauth.py
+
+from datetime import datetime, timezone
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+from jose import JWTError
+import jwt
+from app.settings import settings
+from app.utils import ALGORITHM
+from app.models import User
+from app.api.auth.auth import fake_db_user
+
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/login/")
+
+
+# JWT token verification
+def verify_token(token: str, credentials_exception):
```

Comment:
Do not pass the exception as an argument. If you need to use it in multiple places, define a custom error message and use that instead. However, in your case, you don't need to do that because this exception should only be used for JWT decoding.

Reply:
```
def verify_token(token: str):
    try:
        payload = jwt.decode(token, settings.SECRET_KEY, algorithms=[ALGORITHM])
    
    except JWTError:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Invalid token",
            headers={"WWW-Authenticate": "Bearer"},
        )

    user_id: str = payload.get("sub")
    exp: int = payload.get("exp")
    
    if user_id is None:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Invalid token: subject not found",
            headers={"WWW-Authenticate": "Bearer"},
        )

    if exp is None or datetime.fromtimestamp(exp, tz=timezone.utc) < datetime.now(tz=timezone.utc):
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Token has expired",
            headers={"WWW-Authenticate": "Bearer"},
        )
    
    return user_id
```